src/mine_loop.rs

```diff
@@ -839,19 +845,37 @@ pub(crate) async fn mine(
                 restart_guessing = true;
             }
             Ok(Err(e)) = &mut composer_task => {
-                stop_composing = true;
 
-                match e.downcast_ref::<prover_job::ProverJobError>() {
-                    Some(prover_job::ProverJobError::ProofComplexityLimitExceeded{..} ) => {
-                        pause_mine = true;
-                        tracing::error!("exceeded proof complexity limit.  mining paused.  details: {}", e.to_string())
-                    },
+                match e.root_cause().downcast_ref::<CreateProofError>() {
+                    // address issue 579.
+                    //
+                    // we check if error indicates job was cancelled.
+                    //
+                    // if so, we simply log and continue. ignoring the error.
```

Comment:
```suggestion
                    // Check if error indicates job was cancelled. If so,
                    // simply log and continue, but ignore the error.
```

Reply:
applied suggestion in 1b9a52926cb7755fc853048ec76c5d066a32d47d